File: Java/Medium/Accepted/tuple-with-same-product_3.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        
        int ans = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                int product = nums[i] * nums[j];
                map.put(product, map.getOrDefault(product, 0) + 1);
            }
        }
        for (int product : map.keySet()) {
            int count = map.get(product);
            ans += count * (count - 1) * 4;
        }
        return ans;
    }
}

File: C++/Medium/Accepted/tuple-with-same-product_3.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        
        int n = nums.size();
        int ans = 0;
        unordered_map<int, int> mp;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                mp[nums[i] * nums[j]]++;
            }
        }
        for (auto &x : mp) {
            ans += x.second * (x.second - 1) * 4;
        }
        return ans;
    }
};